lib/puppet/pops/types/types.rb

```diff
@@ -364,6 +402,16 @@ def eql?(o)
 # @api public
 #
 class PType < PTypeWithContainedType
+
+  def self.register_ptype(loader, ir)
+    create_ptype(loader, ir, 'AnyType',
+       'type' => {
+         KEY_TYPE => POptionalType.new(PType::DEFAULT),
+         KEY_VALUE => nil
+       }
+    )
+  end
```

Comment:
No, this type is not abstract. It's a meta-type. No meta-types are abstract.
Not sure what you mean with "repeated". This is a class method that registers the meta-type. The method must be present on all classes that implement such a meta-type.